frontend/device/remarkable/device.lua

```diff
@@ -125,5 +126,159 @@ function Remarkable:reboot()
+        local currentNetwork = getWifiProperty("network")
+        local bsss = getWifiProperty("bSSs")
+        local err = ""
+        if not bsss then
+            return nil, "No network results."
+        end
+        local results = nil
+        for _,path in ipairs(bsss) do
+            if path then
+                local bssid = getBSSProperty(path, "bssid")
+                -- We may need to wait for the object to have been registered properly
+                while not bssid do
+                    os.execute("sleep 1")
+                    bssid = getBSSProperty(path, "bssid")
+                end
+                local ssid = getBSSProperty(path, "ssid")
+                if not isempty(ssid) then
+                    logger.info("Network " .. ssid)
+                    local flags = ""
+                    local keyMgmt = getBSSProperty(path, "key_mgmt")
+                    if keyMgmt then
+                        for _,flag in ipairs(keyMgmt) do
+                            flags = flags .. "[" .. string.upper(flag) .. "]"
+                        end
+                    end
+                    local network = {
+                        bssid = bssid,
+                        ssid = ssid,
+                        frequency = getBSSProperty(path, "frequency"),
+                        signal_level = getBSSProperty(path, "signal"),
+                        flags = flags,
+                        path = path,
+                        connected = false
+                    }
+                    if network.signal then
+                        network.signal_quality = math.min(math.max((network.signal + 100) * 2, 0), 100)
+                    else
+                        network.signal_quality = 0
+                    end
+                    local networkPath = getBSSProperty(path, "network")
+                    if currentNetwork and currentNetwork ~= "/" and networkPath and networkPath == currentNetwork then
+                        network.connected = true
+                    end
+                    if results == nil then
+                        results = {}
+                    end
+                    table.insert(results, network)
+                end
+            else
+                err = err .. "Got nil instead of dbus path. "
+            end
+        end
+        return results, err
+    end
+    function NetworkMgr:getCurrentNetwork()
+        local path = getWifiProperty("network")
+        local results = {}
+        if path and path ~= "/" then
+            table.insert(results, {
+                ssid = getNetworkProperty(path, "ssid"),
+            })
+        end
+        return results
+    end
+    function NetworkMgr:authenticateNetwork(network)
+        local InfoMessage = require("ui/widget/infomessage")
+        local UIManager = require("ui/uimanager")
+        local BD = require("ui/bidi")
+        local info = InfoMessage:new{text = "Connecting to " .. BD.wrap(network.ssid) }
```

Comment:
Is this a temporary debugging aid? The interface has some `Connecting to Wi-Fi…` messages sprinkled around.

If not, it should be something like:

```suggestion
        local info = InfoMessage:new{text = T(_("Connecting to %1.", BD.wrap(network.ssid))) }
```

Reply:
When I was trying to use the template it was never replacing the value.

Reply:
If you mean specifically what I wrote, that'd be because I typod.

Should be:
```lua
local info = InfoMessage:new{text = T(_("Connecting to %1."), BD.wrap(network.ssid)) }
```